crates/hyperqueue/src/common/arraydef.rs

```diff
@@ -104,7 +107,11 @@ impl fmt::Display for IntArray {
                 ));
             }
         }
-        write!(f, "{}", &str[0..str.len() - 2])
+        if str.len() >= 2 {
```

Comment:
We could use `enumerate` and only add `, ` if we're not in the last iteration, that's way less hacky.

Reply:
The method is quite ugly but I have tried to do a minimal touches originally. I have rewritten it in the last commit.